### PR TITLE
auto_crop replace objects param with mask

### PR DESCRIPTION
**Pull request by HaleySchuhl**
**Describe your changes**
A binary mask will replace the `obj` input parameter. Not currently iterable so we would benefit from having a tutorial to show this use case, even though it's been less common. Deep learning is unavoidable lol. 

**Type of update**
Is this a:
* New feature or feature enhancement

**Associated issues**
#945 

**Additional context**
  This function has been used for creating training datasets. This (edge) case usually wants the training to be the same size square so it might require `pcv.roi.rectangle` and `pcv.roi2mask` before creating a labeled mask with `pcv.create_labels`.. probably deserves to be it's own function TBH. Or at least have a tutorial showing this use case since we have known people that want to use pcv for creating DL training sets. 
 
**For the reviewer**
See [this page](https://plantcv.readthedocs.io/en/latest/pr_review_process/) for instructions on how to review the pull request.
- [x] PR functionality reviewed in a Jupyter Notebook
- [x] All tests pass
- [x] Test coverage remains 100%
- [x] Documentation tested
- [x] New documentation pages added to `plantcv/mkdocs.yml`
- [x] Changes to function input/output signatures added to `updating.md`
- [x] Code reviewed
- [x] PR approved

### CODE REVIEW

**Review by nfahlgren — APPROVED** [review]
Nice!
